modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -37,12 +48,31 @@ object GlobalL0Service {
     l0GlobalSnapshotClient: L0GlobalSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
-    singlePullLimit: Option[PosLong]
+    singlePullLimit: Option[PosLong],
+    maybeRealignPeerIds: Option[NonEmptySet[PeerId]] = none
   ): GlobalL0Service[F] =
     new GlobalL0Service[F] {
-
       private val logger = Slf4jLogger.getLogger[F]
 
+      def realignToMajority(lastSnapshot: Hashed[GlobalIncrementalSnapshot]): F[Unit] =
+        maybeRealignPeerIds.traverse_ { realignPeers =>
+          for {
+            peers <- realignPeers.toNonEmptyList
+              .traverse(globalL0ClusterStorage.getPeer)
+              .map(_.toList.flatten)
+
+            majorityHash <- peers
+              .parTraverseN(10)(p => l0GlobalSnapshotClient.getHash(lastSnapshot.ordinal).run(p))
```

Comment:
Seems like single failed response will fail the whole traverse here, if I'm right the question is if that's what you wanted?

Reply:
Is this question resolved? Regarding failures (question that came up during discussion,). During discussion it was mentioned we're filtering failures out and not contributing to majority. I think we should handle cases as follows:

Follow majority excluding failures, implement tie-breaker on event of 1-1-1 split to sort by peer Id for deterministic resolution.